src/main/java/org/candlepin/exceptions/mappers/ServiceExceptionMapper.java

```diff
@@ -46,91 +48,84 @@
-                site.getDellAssetTag());
-        }
-        else if (exception instanceof SubscriptionServiceException) {
-            status = Status.EXPECTATION_FAILED;
-            message = i18n.get().tr(
-                "Unexpected error from Subscription Service: {0}", exception.getMessage());
-        }
-        else if (exception instanceof CloudRegistrationAuthorizationException) {
-            status = Status.UNAUTHORIZED;
-            message = i18n.get().tr(
-                "Cloud provider or account details could not be resolved to an organization");
-        }
-        else if (exception instanceof CloudRegistrationMalformedDataException) {
-            status = Status.BAD_REQUEST;
-            // Some tests return a message. Will retain it instead.
-            message = !StringUtils.isEmpty(message) ? message : i18n.get().tr(
-                "Unable to complete Cloud Registration with provided data");
-        }
-        else if (exception instanceof CloudRegistrationServiceException) {
-            status = Status.BAD_REQUEST;
-            message = i18n.get().tr(
-                "Unexpected error from Cloud Registration Service: {0}", exception.getMessage());
-        }
-        else if (exception instanceof UserInvalidException uie) {
-            status = Status.UNAUTHORIZED;
-            message = i18n.get().tr("User \"{0}\" is not valid.", uie.getUsername());
-        }
-        else if (exception instanceof UserUnacceptedTermsException) {
-            status = Status.UNAUTHORIZED;
-            message = i18n.get().tr("You must first accept Red Hat''s Terms and conditions. " +
-                "Please visit {0} . You may have to log out of and back into the  " +
-                "Customer Portal in order to see the terms.",
-            "https://www.redhat.com/wapps/ugc");
-        }
-        else if (exception instanceof UserDisabledException ude) {
-            status = Status.UNAUTHORIZED;
-            message = i18n.get().tr("The user \"{0}\" has been disabled, if this is a mistake, " +
-                "please contact customer service.", ude.getUsername());
-        }
-        else if (exception instanceof UserUnauthorizedException) {
-            status = Status.UNAUTHORIZED;
-            message = i18n.get().tr("Invalid username or password. To create a login, " +
-                "please visit {0}", "https://www.redhat.com/wapps/ugc/register.html");
-        }
-        else if (exception instanceof UserServiceException) {
-            status = Status.UNAUTHORIZED;
-            message = i18n.get().tr("Unexpected error from User Service: {0}", exception.getMessage());
-        }
-        else if (exception instanceof ProductServiceException pse) {
-            status = Status.BAD_REQUEST;
-            message = i18n.get().tr("Unable to retrieve proudct \"{0}\" from ProductService",
-                pse.getProductId());
+        Pair<Status, String> result = mappings.get(exception.getClass());
```

Comment:
Note that this lookup doesn't work with subclasses.

When/if we move to Java 21, we can start using the switch expressions to best wrap all of this up in a single mapping function, but for now we may need to either do a semi-recursive lookup, or at least document that subclasses of registered mappings aren't implicitly handled as one might expect.

@Januson thoughts on this?

Reply:
I agree. Switch would make it a bit more concise, but until then, pattern matching with instanceOf should be good enough.